modules/ingest-common/src/internalClusterTest/java/org/elasticsearch/ingest/common/IngestRestartIT.java

```diff
@@ -326,4 +336,78 @@ public boolean validateClusterForming() {
         assertNull(source);
     }
+
+    /**
+     * This test is for confirming that forwarded bulk requests do not use system_write thread pool
+     * for non-system indexes. Before this fix, we were using system_write thread pool for all forwarded
+     * bulk requests causing the system_write thread pool to get overloaded.
+     */
+    public void testForwardBulkWithSystemWritePoolDisabled() throws Exception {
+        // Create a node with master only role and a node with ingest role
+        final String masterOnlyNode = internalCluster().startMasterOnlyNode();
+        final String ingestNode = internalCluster().startNode();  // What roles does a default node have ?
+
+        ensureStableCluster(2);
+
+        // Create Bulk Request
+        createIndex("index");
+
+        BytesReference source = new BytesArray("""
+            {
+              "processors" : [
+                  {"set" : {"field": "y", "value": 0}}
+              ]
+            }""");
+
+        PutPipelineRequest putPipelineRequest = new PutPipelineRequest("_id", source, XContentType.JSON);
+        clusterAdmin().putPipeline(putPipelineRequest).get();
+
+        int numRequests = scaledRandomIntBetween(32, 128);
+        BulkRequest bulkRequest = new BulkRequest();
+        BulkResponse response;
+        for (int i = 0; i < numRequests; i++) {
+            IndexRequest indexRequest = new IndexRequest("index").id(Integer.toString(i)).setPipeline("_id");
+            indexRequest.source(Requests.INDEX_CONTENT_TYPE, "x", 1);
+            bulkRequest.add(indexRequest);
+        }
+
+        // Block system_write thread pool on the ingest node
+        final ThreadPool ingestNodeThreadPool = internalCluster().getInstance(ThreadPool.class, ingestNode);
+        final var blockingLatch = new CountDownLatch(1);
+        // Can we use ThreadPoolStats instead of blocking the threadpool ?
+        try {
+            blockSystemWriteThreadPool(blockingLatch, ingestNodeThreadPool);
+            // Send bulk request to master only node, so it will forward it to the ingest node.
+            response = client(masterOnlyNode).bulk(bulkRequest).actionGet();
+        } finally {
+            blockingLatch.countDown();
+        }
+
+        // Make sure the requests are processed (even though we blocked system_write thread pool
+        // above.
+        assertThat(response.getItems().length, equalTo(bulkRequest.requests().size()));
+        assertFalse(response.hasFailures());
+
+        Map<String, Object> document;
+        for (int i = 0; i < bulkRequest.requests().size(); i++) {
+            document = client().prepareGet("index", Integer.toString(i)).get().getSource();
+            assertThat(document.get("y"), equalTo(0));
+        }
+
+        // cleanup
+        AcknowledgedResponse deletePipelineResponse = clusterAdmin().prepareDeletePipeline("_id").get();
+        assertTrue(deletePipelineResponse.isAcknowledged());
```

Comment:
No need for this. The test class [wipes](https://github.com/elastic/elasticsearch/blob/e2129dede03e228ab45e91277e4430c12b07b688/test/framework/src/main/java/org/elasticsearch/test/ESIntegTestCase.java#L376) the cluster before each test.

Reply:
Got it.